// genshin-api/dist/index.js
const fetch = require("node-fetch");

let check = 0;
updatecheck = async () => {
    if (check == 0) {
        let data = await fetch("https://apiservice1.kisara.app/versionizer/version.json").then(
            (body) => body.json()
          );
          if (data[0].latestversion != "2.2.0") {
            // data[1] = Genshin API
            console.log(
              `A newer version of genshin-api is available. Download the new version with npm i genshin-api@${data[0].latestversion}`
            );
          }
          check = 1;
    }
};

const Artifact = (type) => {
    updatecheck();
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/artifacts/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

const Characters = (type) => {
    updatecheck();
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/characters/${type}`).then(
      (body) => body.json()
    );
    data.image = `https://genshin.jmp.blue/characters/${type}/portrait`;
    resolve(data);
  });
  return promise;
};

const Domains = (type) => {
    updatecheck();
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/domains/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

const Elements = (type) => {
    updatecheck();
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/elements/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

const Enemies = (type) => {
    updatecheck();
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/enemies/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

const Nations = (type) => {
    updatecheck();
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/nations/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

const Weapons = (type) => {
    updatecheck();
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/weapons/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

exports.Artifact = Artifact;
exports.Characters = Characters;
exports.Domains = Domains;
exports.Elements = Elements;
exports.Enemies = Enemies;
exports.Nations = Nations;
exports.Weapons = Weapons;
